File: leetcode_py/Count Primes.py
import math
class Solution:
    # @param {integer} n
    # @return {integer}
    def countPrimes(self, n):
        if n<3:
            return 0
        flag=[0,0]+ [1] * (n-2)
        
        end=int(math.sqrt(n))+1
        for i in range(2,end):
            if flag[i]==1:
                j=i
                while j+i<n:
                    j+=i
                    flag[j]=0
        return sum(flag)
  
    
if __name__=="__main__":
    a=Solution()
    print a.countPrimes(2)     
    print a.countPrimes(5)           

File: leetcode_js/Count Primes.js
/**
 * @param {number} n
 * @return {number}
 */
var countPrimes = function(n) {
    var res = new Array(n);
    for(var i = 2; i < n; i++) res[i] = 1;
    for(var i = 2; i < Math.floor(Math.sqrt(n))+1; i++){
        for(var j = i*2; j < n; j += i){
            res[j] = 0;
        }
    }
    var re = 0;
    for(i = 0; i < n; i++){
        if(res[i] === 1) re += 1;
    }
    return re;
};